lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -183,24 +183,20 @@ def test_bad_dates(self, client, server_config, user_ok, pbench_token):
         """
         Test behavior when a bad date string is given
         """
-        skip = True
         for key, p in self.cls_obj.schema.parameters.items():
             # Modify date/time key in the payload to make it look invalid
             if p.type == ParamType.DATE and key in self.payload:
                 self.payload[key] = "2020-19"
-                skip = False
-        if skip:
-            pytest.skip("skipping " + self.test_bad_dates.__name__)
-        response = client.post(
-            server_config.rest_uri + self.pbench_endpoint,
-            headers={"Authorization": "Bearer " + pbench_token},
-            json=self.payload,
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert (
-            response.json.get("message")
-            == "Value '2020-19' (str) cannot be parsed as a date/time string"
-        )
+                response = client.post(
+                    server_config.rest_uri + self.pbench_endpoint,
+                    headers={"Authorization": "Bearer " + pbench_token},
+                    json=self.payload,
+                )
+                assert response.status_code == HTTPStatus.BAD_REQUEST
+                assert (
+                    response.json.get("message")
+                    == "Value '2020-19' (str) cannot be parsed as a date/time string"
+                )
```

Comment:
If I understand this code, it's going to look for a `DATE` field, and, if it finds one, it's going to issue the request; if that behaves properly, then it's going to continue looking for another `DATE` field.  While it's looking, it won't issue any requests until it finds one (which is fine); when it finds one, it's going to damage the date value (which is fine -- that's how it drives the test); however, at the point when it's going to issue the second request, it's still got the damaged value from the first request in the payload!  So, the second request is unlikely to test anything different from the first one.  😿 

In general, I think I like this approach.  But, I think it needs to save and restore the original value after the assertions succeed and before it moves to the next iteration.

Reply:
I understand what you mean, but the api code is gonna behave exactly same for any field that is of `DATE` type (because it passes the validation via one function, `conver_date`) so in practice really if this test works for one `Date` field we dont need to test on any other `DATE` field, it would anyway be redundant. However, I guess it wouldn't hurt to test bad date on each `Date` field separately since we are already inside a for loop

Reply:
> the api code is gonna behave exactly same for any field that is of `DATE` type (because it passes the validation via one function, `conver_date`)

Perhaps that's true today, but the unit test code doesn't (and shouldn't!) know that, because it might not be true tomorrow.  The unit tests should verify the "interface contract", which (IMO) says that, if any `DATE` field is invalid, the API will respond in a particular way...and that means it should test them all, not just the first one (especially since it was easy to make it do so 😉).